plutus-numeric/src/PlutusTx/SchemaUtils.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+
+module PlutusTx.SchemaUtils (
+  RatioDirection (..),
+  ratioDeclareNamedSchema,
+  ratioFixFormArgument,
+  ratioFormSchema,
+  ratioTypeName,
+  jsonFieldSym,
+) where
+
+import Data.Functor.Foldable (Fix (Fix))
+import Data.OpenApi qualified as OpenApi
+import Data.OpenApi.Declare (type Declare)
+import Data.Proxy (Proxy (Proxy))
+import Data.Text (Text, pack)
+import Data.Text qualified as Text
+import GHC.Exts
+import GHC.TypeLits (KnownSymbol, Symbol, symbolVal)
+import Schema (
+  FormArgumentF (FormObjectF),
+  FormSchema (FormSchemaObject),
+  toArgument,
+  toSchema,
+ )
+import Prelude
+
+{- | Type-level data representing the "direction" that a ratio converts to or from.
+
+ @since 1.2
+-}
+data RatioDirection = Symbol :-> Symbol
+
+-- | @since 1.2
```

Comment:
This function isn't very clearly named: its goal is to turn a `KnownSymbol s => s` into `Text`. Maybe `symbolToText`?

Reply:
Now RatioDirection is used for specifying direction of transformation:
`a * ratio = b` corresponds to `symbol_a :-> symbol_b`

I don't like this, because of flipped symbols order: `denominator :-> numerator`

Maybe to do this way: `data RatioFields = Symbol :-> Symbol` (numerator :-> denominator)

Reply:
I agree - the original direction always confused me. If we do this, we need to make sure that the derived instances in `liqwid-contracts` are changed to match.

Reply:
Сompletely changed the given type